frontend/src/main.ts

```diff
@@ -5,28 +5,39 @@ import VueMaterial from 'vue-material';
 
 import Vue from 'vue';
 import App from './App.vue';
-import router from './router';
+import router, { isErrorPage, goToMainPage, goToUnlockMetamaskPage, goToUnsupportedNetworkPage, goToInstallMetamaskPage, goToErrorPage } from './router';
 import API from './api';
 
 import { VueConstructor } from 'vue/types/vue';
 import registerPurpleTheme from './theme';
 
 Vue.use(VueMaterial);
-Vue.use(VueMaterial.MdIcon);
-
-window.addEventListener('load', async () => {
-  API.init();
-});
 
 registerPurpleTheme(Vue);
 
 Vue.config.productionTip = false;
 
 
 /* eslint-disable no-new */
-new Vue({
+var _Vue = new Vue({
   el: '#app',
   router,
   template: '<App/>',
   components: { App },
 });
+
+const actions = {
+  metamaskNotInstalled: () => goToInstallMetamaskPage.call(_Vue),
+  metamaskLocked: () => goToUnlockMetamaskPage.call(_Vue),
+  unsupportedNetwork: () => goToUnsupportedNetworkPage.call(_Vue),
+  unknownError: () => goToErrorPage.cal(_Vue),
```

Comment:
That `.cal` here looks like a typo. It should throw the compiler off, consider adding more types? :)

Reply:
You can probably also do
`metamaskNotInstalled: goToInstallMetamaskPage.apply(_Vue)`